api/src/auth/auth.controller.ts

```diff
@@ -1,37 +1,157 @@
-import { Controller, Post, UseGuards, Request, Response, Get } from '@nestjs/common';
+import { Controller, UseGuards, Request, Get, Res, UseFilters } from '@nestjs/common';
 import { AuthService } from './auth.service';
-import { ApiOkResponse, ApiTags } from '@nestjs/swagger';
+import { ApiBody, ApiDefaultResponse, ApiInternalServerErrorResponse, ApiOkResponse, ApiQuery, ApiTags, ApiUnauthorizedResponse } from '@nestjs/swagger';
 import { LocalAuthGuard } from './local-auth.guard';
-import { Response as ResponseType } from 'express';
-import { ConnectUsersDto } from 'src/users/dto/connect-user.dto';
+import { Response, Response as ResponseType } from 'express';
 import { AppUser } from 'src/types/clientSchema';
 import { JwtAuthGuard } from './jwt-auth.guard';
 import { UsersService } from 'src/users/users.service';
+import { OauthGuard } from './oauth.guard';
+import { HttpExceptionFilter } from './redirect-exception.filter';
 
 @ApiTags('authentification')
 @Controller('auth')
 export class AuthController {
   constructor(private readonly authService: AuthService, private readonly usersService: UsersService) {}
 
+  //#region oauth
+
+  /**
+   * GET /api/auth/oauth
+   * This is the route the user will visit to authenticate using the 42 API
+   */
+  @ApiDefaultResponse({description: "Nothing to return, the user will be redirected to the 42 API"})
+  @Get('oauth')
+  @UseGuards(OauthGuard)
+  login() {
+    return ;
+  }
+
+  //#endregion
+
+  //#region redirect
+
+  /**
+   * GET /api/auth/redirect
+   * This is the route the user will visit to get a JWT Token after visiting 42,
+   * We could do everything using just the /api/auth/oauth route. But this is less
+   * complex to understand.
+   */
+  @ApiQuery({
+    name: 'code',
+    required: true,
+    description: "User token of 42 API. Note: This endpoint should not be use directly, as you will never get a valid token for the 42 API. This endpoint purpose is solely for the 42 API."
+  })
+  @ApiOkResponse({
+    description: "This will redirect to either localhost:3000/ if a jwt as been generated, or localhost:3000/login?code=x if something went wrong. Note that x will be replaced with a string of the error."
+  }) 
+  @Get('redirect')
+  @UseGuards(OauthGuard)
+  @UseFilters(HttpExceptionFilter)
+  async redirect(@Request() req, @Res() res: Response) {
+    const token: string = await this.authService.login(req.user.id);
+    let date = new Date();
+    date.setDate(date.getDate() + 7);
+    res.cookie('token', token, {expires: date});
+    res.redirect('http://localhost:3000/');
```

Comment:
It would be better to place the string inside a `.env`. Or at least load the port from a .env

Reply:
Commit dae2f0e use the url from a env var

Reply:
Commit dae2f0e use the url from a env var